lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,155 @@
+        onChange?.(e);
+    };
+
+    const handleIconClick = (
+        e:
+            | React.MouseEvent<HTMLSpanElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => {
+        if (!disabled) {
+            const synthesizedEvent = new Event("change");
+            const input = e.currentTarget.previousSibling as HTMLInputElement;
+            input.dispatchEvent(synthesizedEvent);
+        }
+    };
+
+    const getIcon = () => {
+        if (checked) {
+            return size === "sm" ? (
+                <LabelPairedCircleDotSmFillIcon
+                    data-testid={`dt_checked_icon_${value}_${disabled}`}
+                    fill={disabled ? "#b8b8b8" : "#000000"}
+                    height={22}
+                    width={22}
+                />
+            ) : (
+                <LabelPairedCircleDotMdFillIcon
+                    data-testid={`dt_checked_icon_${value}_${disabled}`}
+                    fill={disabled ? "#b8b8b8" : "#000000"}
+                    height={24}
+                    width={24}
+                />
+            );
+        } else {
+            return size === "sm" ? (
+                <LabelPairedCircleSmRegularIcon
+                    data-testid={`dt_unchecked_icon_${value}_${disabled}`}
+                    fill={disabled ? "#b8b8b8" : "#7e7e7e"}
+                    height={22}
+                    width={22}
+                    tabIndex={0}
+                />
+            ) : (
+                <LabelPairedCircleMdRegularIcon
+                    data-testid={`dt_unchecked_icon_${value}_${disabled}`}
+                    fill={disabled ? "#b8b8b8" : "#7e7e7e"}
+                    height={24}
+                    width={24}
+                    tabIndex={0}
+                />
+            );
+        }
+    };
+
+    return (
+        <label
+            htmlFor={id}
+            className={clsx("quill-radio-button", className, {
+                "quill-radio-group__item--active": checked,
+            })}
+            data-testid="dt_quill_radio_button"
+        >
+            <input
+                className="quill-radio-button__input"
+                type="radio"
+                onChange={onInputChange}
+                checked={checked}
+                disabled={disabled}
+                value={value}
+                {...otherProps}
+            />
+
+            <span
+                className="quill-radio-button__icon"
+                onClick={handleIconClick}
+                onKeyDown={handleIconClick}
+                id={id}
+            >
+                {getIcon()}
+            </span>
+            <Text
+                size={size}
+                as="span"
+                className={clsx("quill-radio-button__label", classNameLabel)}
+            >
+                {children}
+            </Text>
+            {/* TODO: implement info icon component */}
+            {has_info && (
+                size === "sm" ? (
```

Comment:
Can replace with:
 ```
       const InfoIcon =
            size === "sm"
                ? LabelPairedCircleInfoSmRegularIcon
                : LabelPairedCircleInfoMdRegularIcon;
                
                ....
                render (
                ....
                {hasInfo &&  <InfoIcon
                        className={clsx(
                            "quill-radio-button__info",
                            classNameInfo,
                        )}
                        data-testid="dt_quill_radio_button_info"
                    />}
                    ...)
```

Reply:
Why? There are several ways to achieve the same thing in programming. It's not an exact science...